app/models/course.rb

```diff
@@ -117,6 +119,31 @@ def root_folder?
     end
   end
 
+  def initialize_duplicate(duplicator, other)
+    self.start_at += duplicator.time_shift
+    self.end_at += duplicator.time_shift
+    self.title = duplicator.new_course_title
+
+    # This also duplicates assessments.
+    self.lesson_plan_items = duplicator.duplicate(other.lesson_plan_items.map(&:actable)).
```

Comment:
Would it be better to just duplicate the `lesson_plan_items`? Then we handle the different types of `lesson_plan_items` within the `initialize_duplicate` method for `lesson_plan_items`.